File: Code/PharmacyInformationSystem/src/businessLogic/loginFunctionality/AuthenticationFailure.java
package businessLogic.loginFunctionality;

public class AuthenticationFailure extends Throwable{

	private String message;
	
	/*Constructor*/
	public AuthenticationFailure(String message) throws Exception {
		super();
		this.message = message;
		throw new Exception(message);
	}
	
	/*Setter n Getters*/
	public String getMessage() {
		return message;
	}

	public void setMessage(String message) {
		this.message = message;
	}
	
}

File: Code/PharmacyInformationSystem/PharmacyInformationSystem/BusinessLogic/LoginFunctionality/AuthenticationFailure.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PharmacyInformationSystem.BusinessLogic.LoginFunctionality
{
    /// <summary>
    /// Exception thrown when the authentication fails
    /// Contains a Message with the reason
    /// </summary>
    public class AuthenticationFailure : System.Exception
    {
        public AuthenticationFailure(string message): base(message) { }
    }
}
